feat: add C implementation `stats/base/dists/beta/mgf` (#4779)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/beta/mgf/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/beta/mgf.h"
```

#### stdlib_base_dists_beta_mgf( t, alpha, beta )

> Returns the [mgf][mgf] of a [beta][beta-distribution] distribution with parameters `alpha` (first shape parameter) and `beta` (second shape parameter) with input value `t`.

Reviewer: ```suggestion
Evaluates the [moment-generating function][mgf] (MGF) for a [beta][beta-distribution] distribution with parameters `alpha` (first shape parameter) and `beta` (second shape parameter).
```

```c
double y = stdlib_base_dists_beta_mgf( 0.5, 1.0, 1.0 );
// returns ~1.297
```

The function accepts the following arguments:

-   **t**: `[in] double` input value.
-   **alpha**: `[in] double` first shape parameter.
-   **beta**: `[in] double` second shape parameter.

```c
double stdlib_base_dists_beta_mgf( const double t, const double alpha, const double beta );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/beta/mgf.h"
#include "stdlib/constants/float64/eps.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double t;
    double alpha;
    double beta;
    double y;

>     int i;

Reviewer: ```suggestion
    double alpha;
    double beta;
    double t;
    double y;
    int i;
```

    for ( i = 0; i < 25; i++ ) {
        t = random_uniform( 0.0, 20.0 );
        alpha = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
        beta = random_uniform( 1.0 + STDLIB_CONSTANT_FLOAT64_EPS, 100.0 );
        y = stdlib_base_dists_beta_mgf( alpha, beta );

>         printf( "t: %lf, alpha: %lf, beta: %lf, mgf(X;t,alpha,beta): %lf\n", t alpha, beta, y );

Reviewer: ```suggestion
        printf( "t: %lf, α: %lf, β: %lf, M_X(t;α,β): %lf\n", t, alpha, beta, y );
```
